asset-transfer-private-data/application-javascript/app_org1.js

```diff
@@ -0,0 +1,144 @@
+    const caTLSCACerts = caInfo.tlsCACerts.pem;
+    const caService = new FabricCAServices(caInfo.url, { trustedRoots: caTLSCACerts, verify: false }, caInfo.caName);
+    const mspOrg1 = "Org1MSP"; 
+
+    // Steps:
+    // Note: Steps 1 & 2 need to done only once in an app-server per blockchain network
+    // 1. register & enroll admin user with CA, stores admin identity in local wallet
+    await caUtils.EnrollOrgAdminUser(mspOrg1, walletOrg1, caService);
+
+    // 2. register & enroll application user with CA, which is used as client identify to make chaincode calls, stores app user identity in local wallet
+    await caUtils.RegisterOrgUser(caUtils.Org1UserId, mspOrg1, walletOrg1, caService);
+
+    try {
+        // Create a new gateway for connecting to Org's peer node.
+        const gatewayOrg1 = new Gateway();
+        await gatewayOrg1.connect(ccpOrg1,
+            { wallet: walletOrg1, identity: caUtils.Org1UserId, discovery: { enabled: true, asLocalhost: true } });
+        
+        return gatewayOrg1;
+    } catch (error) {
+        console.error(`Error in connecting to gateway: ${error}`);
+        process.exit(1);
+    }
+}
+
+
+// See this usecase details at asset-transfer-private-data/chaincode-go/README.md
+// Uses for fabric-samples/test-network based setup and the companion chaincode
+async function main() {
+    try {
+        // For this usecase illustration, we will use both Org1 & Org2 client from this same app
+        // In real world this will be different apps
+
+        /** ******* Fabric client init: Using Org1 identity to Org1 Peer ********** */
+        const gatewayOrg1 = await initContractFromOrg1Identity();
+        const networkOrg1 = await gatewayOrg1.getNetwork(myChannel);
+        const contractOrg1 = networkOrg1.getContract(myChaincodeName);
+        
+        try {
+            
+            // *** Some example transactions are listed below ***
+
+            // Add new sample Assets & use transfer one of the asset to new owner
+            let assetID1 = "asset1";
+            let assetID2 = "asset2";
+            const assetType = "asset";
+            let result;
+            let asset1Data = {objectType: assetType, assetID: assetID1, color: "green", size: 20,  appraisedValue: 100};
+            let asset2Data = {objectType: assetType, assetID: assetID2, color: "blue", size: 35,  appraisedValue: 727};
+
+
+            console.log('\n**************** As Org1 Client ****************');            
+            console.log('Adding Assets to work with: Submit Transaction: CreateAsset '+assetID1);
+            console.log("todo Adding " + JSON.stringify(asset1Data));
+            let statefulTxn = contractOrg1.createTransaction('CreateAsset');
+            let tmapData = Buffer.from(JSON.stringify(asset1Data));
+            statefulTxn.setTransient( {
+                asset_properties: tmapData
+            });
+            result = await statefulTxn.submit();                        
```

Comment:
note: using Org1 Identity

Intermitted error (since every other or so, of the node sdk  is sending the Txn Invokes to Org2 peer)

2020-07-17T13:18:26.154Z - error: [Transaction]: Error: No valid responses from any peers. Errors:
    peer=peer0.org1.example.com:7051, status=500, message=failed to put asset bid: collection must not be an empty string